frontends/p4/actionsInlining.cpp

```diff
@@ -84,13 +80,14 @@ const IR::Node *ActionsInliner::preorder(IR::MethodCallStatement *statement) {
     substitution.populate(callee->parameters, statement->methodCall->arguments);
 
     // evaluate in and inout parameters in order
-    for (auto param : callee->parameters->parameters) {
-        auto argument = substitution.lookup(param);
-        cstring newName = refMap->newName(param->name.name.string_view());
+    for (const auto *param : callee->parameters->parameters) {
+        const auto *argument = substitution.lookup(param);
+        cstring newName =
+            nameGen.newName(param->name.string() + "_inlined_" + callee->name.string());
```

Comment:
Just a quick skim for now, what is the deal with these renames? Why add this renaming?

Reply:
We need to rename parameters in any way. This gives a bit more indicative base name after inlining (e.g. when caller had `x` and callee has `x` we'll end with `x` and `x_inlined_callee` instead of `x` and `x0`). This was mostly for my debugging, but I decided to leave it as-is.

Reply:
I know I request this a lot but could we split it out into a separate if it's unrelated to the ref map change? Considering it's a breaking change we should remove any confounding factors.

Reply:
Well, it would still be a change. As instead of `refMap` we're using name generator here.

Reply:
Ah I see, doesn't seem like there is a way to preserve the naming here. The counters are different. Preserving the name would remove a lot of noise.

Also I recommend making this `IR:ID` and preserve the originalName in the generator. The previous implementation did not do this either but it is useful to have in downstream compiler passes.

Reply:
We can return to the old "scheme" actually so the counters would match, let me do this. As for preserving `originalName` – I'd do this in the separate issue. See the "take home messages" above. We do recalculate name generators very often. In the testcase above each sweep of NameGenerator over the whole program costs 50 ms (!) due to visitor malloc traffic and GC. This totals to ~6 seconds of overall runtime (yes, we do recalculate that often).

Instead we should have some central `SymbolTable`. And all `IR:ID` / name generation should be routed through it.

Reply:
So, I've returned old scheme, plus did few improvements to reduce number of `NameGenerator` sweeps.

Reply:
Thanks! We can add your naming scheme in a follow-up PR if you think it is better. I do not see the harm, any tool depending on an internal name is likely doing something wrong here.